manifests/charts/istio-control/istio-discovery/templates/configmap.yaml

```diff
@@ -9,17 +9,17 @@
     # is processed as if it were declared in the leaf namespace.
     rootNamespace: {{ .Values.meshConfig.rootNamespace | default .Values.global.istioNamespace }}
 
-  {{ $prom := include "prometheus" . | eq "true" }}
-  {{ $sdMetrics := include "sd-metrics" . | eq "true" }}
-  {{ $sdLogs := include "sd-logs" . | eq "true" }}
+  {{ $prom := include "default-prometheus" . | eq "true" }}
+  {{ $sdMetrics := include "default-sd-metrics" . | eq "true" }}
+  {{ $sdLogs := include "default-sd-logs" . | eq "true" }}
   {{- if or $prom $sdMetrics $sdLogs }}
     defaultProviders:
     {{- if or $prom $sdMetrics }}
       metrics:
       {{ if $prom }}- prometheus{{ end }}
-      {{ if $sdMetrics }}- stackdriver{{ end }}
+      {{ if and $sdMetrics $sdLogs }}- stackdriver{{ end }}
```

Comment:
why do we need logs enabled to turn on metrics? And the opposite on L22

Reply:
The semantics would be: only set defaultProviders if both metrics and logs are default.
If one of them is default and another is custom, say metrics is default and log has custom setting, and we write {{ if $sdMetrics }}, it will end up having both defaultProviders set and EnvoyFilters created. 

Not sure if it is okay to have both, but I think having only one of them can reduce confusion.